File: sdks/go/examples/basic_usage.go
package main

import (
	"context"
	"fmt"
	"log"

	vaultagent "github.com/vault-agent/go-sdk"
)

func main() {
	// Initialize client with API key authentication
	auth := vaultagent.NewAPIKeyAuth("your-api-key-here")
	
	client, err := vaultagent.NewClient(
		"https://localhost:8200",
		auth,
		vaultagent.WithTimeout(30*time.Second),
		vaultagent.WithCache(true, 5*time.Minute, 1000),
		vaultagent.WithLogLevel("info"),
	)
	if err != nil {
		log.Fatalf("Failed to create client: %v", err)
	}
	defer client.Close()

	ctx := context.Background()

	// Create a secret
	secret, err := client.CreateSecret(ctx, vaultagent.CreateSecretRequest{
		Name:  "database-password",
		Value: "super-secret-password",
		Metadata: map[string]interface{}{
			"environment": "production",
		},
		Tags: []string{"database", "production"},
	})
	if err != nil {
		log.Fatalf("Failed to create secret: %v", err)
	}
	fmt.Printf("Created secret: %s\n", secret.ID)

	// Retrieve the secret
	retrieved, err := client.GetSecret(ctx, secret.ID)
	if err != nil {
		log.Fatalf("Failed to get secret: %v", err)
	}
	fmt.Printf("Retrieved secret value: %s\n", retrieved.Value)

	// List secrets
	secrets, err := client.ListSecrets(ctx, vaultagent.ListSecretsOptions{
		Tags: []string{"production"},
	})
	if err != nil {
		log.Fatalf("Failed to list secrets: %v", err)
	}
	fmt.Printf("Found %d production secrets\n", len(secrets))

	// Update secret
	newValue := "new-password"
	updated, err := client.UpdateSecret(ctx, secret.ID, vaultagent.UpdateSecretRequest{
		Value: &newValue,
		Metadata: map[string]interface{}{
			"environment": "production",
			"updated":     "true",
		},
	})
	if err != nil {
		log.Fatalf("Failed to update secret: %v", err)
	}
	fmt.Printf("Updated secret to version %d\n", updated.Version)

	// Health check
	health, err := client.HealthCheck(ctx)
	if err != nil {
		log.Fatalf("Failed to check health: %v", err)
	}
	fmt.Printf("Vault status: %s\n", health.Status)
}

File: sdks/nodejs/examples/basic-usage.js
/**
 * Basic usage example for Vault Agent Node.js SDK
 */

const { VaultAgentClient, APIKeyAuth } = require('@vault-agent/sdk');

async function main() {
  // Initialize client with API key authentication
  const auth = new APIKeyAuth('your-api-key-here');
  
  const client = new VaultAgentClient('https://localhost:8200', auth, {
    timeout: 30000,
    cacheEnabled: true,
    logLevel: 'info'
  });

  try {
    // Create a secret
    const secret = await client.createSecret({
      name: 'database-password',
      value: 'super-secret-password',
      metadata: { environment: 'production' },
      tags: ['database', 'production']
    });
    console.log(`Created secret: ${secret.id}`);

    // Retrieve the secret
    const retrieved = await client.getSecret(secret.id);
    console.log(`Retrieved secret value: ${retrieved.value}`);

    // List secrets
    const secrets = await client.listSecrets({ tags: ['production'] });
    console.log(`Found ${secrets.length} production secrets`);

    // Update secret
    const updated = await client.updateSecret(secret.id, {
      value: 'new-password',
      metadata: { environment: 'production', updated: 'true' }
    });
    console.log(`Updated secret to version ${updated.version}`);

    // Health check
    const health = await client.healthCheck();
    console.log(`Vault status: ${health.status}`);

  } catch (error) {
    console.error('Error:', error.message);
  } finally {
    client.close();
  }
}

main().catch(console.error);